src/api/operator/numpy/np_matrix_op.cc

```diff
@@ -467,4 +467,24 @@ MXNET_REGISTER_API("_npi.diag_indices_from")
   *ret = ndoutputs[0];
 });
 
+MXNET_REGISTER_API("_npi.vstack")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  const nnvm::Op* op = Op::Get("_npi_vstack");
+  nnvm::NodeAttrs attrs;
+  op::NumpyVstackParam param;
+  param.num_args = args.size();
+
+  attrs.parsed = param;
+  attrs.op = op;
+  SetAttrDict<op::NumpyVstackParam>(&attrs);
+  int num_outputs = 0;
+  std::vector<NDArray*> inputs;
+  for (int i = 0; i < param.num_args; ++i) {
+    inputs.push_back(args[i].operator mxnet::NDArray*());
+  }
+  auto ndoutputs = Invoke(op, &attrs, param.num_args, &inputs[0], &num_outputs, nullptr);
```

Comment:
Please avoid `inputs[0]` because there is no guarantee that it ever exists. I think there will be some standard (and safer) ways to get the address of vector data.